pkg/argocd/argocd.go

```diff
@@ -25,13 +25,21 @@ import (
 
 // Kubernetes based client
 type k8sClient struct {
-	kubeClient *kube.ImageUpdaterKubernetesClient
+	kubeClient   *kube.ImageUpdaterKubernetesClient
+	appNamespace *string
 }
 
-// GetApplication retrieves an application by name across all namespaces.
+// GetApplication retrieves an application by name, either in a specific namespace or all namespaces depending on client configuration.
 func (client *k8sClient) GetApplication(ctx context.Context, appName string) (*v1alpha1.Application, error) {
-	// List all applications across all namespaces (using empty labelSelector)
-	appList, err := client.ListApplications(v1.NamespaceAll)
+	// List all applications across configured namespace or all namespaces (using empty labelSelector)
+	if *client.appNamespace != v1.NamespaceAll {
+		return client.kubeClient.ApplicationsClientset.ArgoprojV1alpha1().Applications(*client.appNamespace).Get(ctx, appName, v1.GetOptions{})
+	}
+	return client.getApplicationInAllNamespaces(ctx, appName)
+}
+
+func (client *k8sClient) getApplicationInAllNamespaces(ctx context.Context, appName string) (*v1alpha1.Application, error) {
```

Comment:
the param `ctx` is not used in this new function. I think we should remove it.

Reply:
[eab19bd](https://github.com/argoproj-labs/argocd-image-updater/pull/1112/commits/eab19bdef06599e643a5f0613c537bf50c6c88ff)